stake-pool/program/src/instruction.rs

```diff
@@ -1323,6 +1323,64 @@ pub fn decrease_validator_stake_with_vote(
     )
 }
 
+/// Create a `IncreaseAdditionalValidatorStake` instruction given an existing
+/// stake pool, valiator list and vote account
+pub fn increase_additional_validator_stake_with_list(
+    program_id: &Pubkey,
+    stake_pool: &StakePool,
+    validator_list: &ValidatorList,
+    stake_pool_address: &Pubkey,
+    vote_account_address: &Pubkey,
+    lamports: u64,
+    ephemeral_stake_seed: u64,
+) -> Instruction {
+    let validators = validator_list
+        .find(vote_account_address)
+        .ok_or(ProgramError::InvalidInstructionData)
+        .unwrap();
+    let transient_stake_seed = u64::from(validators.transient_seed_suffix);
+    let validator_stake_seed = NonZeroU32::new(validators.validator_seed_suffix.into());
+    increase_additional_validator_stake_with_vote(
+        program_id,
+        stake_pool,
+        stake_pool_address,
+        vote_account_address,
+        lamports,
+        validator_stake_seed,
+        transient_stake_seed,
+        ephemeral_stake_seed,
+    )
+}
+
+/// Create a `DecreaseAdditionalValidatorStake` instruction given an existing
+/// stake pool, valiator list and vote account
+pub fn decrease_additional_validator_stake_with_list(
+    program_id: &Pubkey,
+    stake_pool: &StakePool,
+    validator_list: &ValidatorList,
+    stake_pool_address: &Pubkey,
+    vote_account_address: &Pubkey,
+    lamports: u64,
+    ephemeral_stake_seed: u64,
+) -> Instruction {
+    let validators = validator_list
+        .find(vote_account_address)
+        .ok_or(ProgramError::InvalidInstructionData)
+        .unwrap();
```

Comment:
Same here, let's return an error and rename the variable
```suggestion
) -> Result<Instruction, ProgramError> {
    let validator_info = validator_list
        .find(vote_account_address)
        .ok_or(ProgramError::InvalidInstructionData)?;
```